src/librustdoc/formats/cache.rs

```diff
@@ -244,10 +245,18 @@ impl<'a, 'tcx> DocFolder for CacheBuilder<'a, 'tcx> {
             }
         }
 
+        let tcx = self.tcx;
         // Propagate a trait method's documentation to all implementors of the
         // trait.
         if let clean::TraitItem(ref t) = *item.kind {
-            self.cache.traits.entry(item.def_id).or_insert_with(|| t.clone());
+            // FIXME: We don't use the returned `&mut` so better not use the `entry` API here.
+            self.cache.traits.entry(item.def_id).or_insert_with(|| clean::TraitWithExtraInfo {
+                trait_: t.clone(),
+                is_spotlight: clean::utils::has_doc_flag(
+                    tcx.get_attrs(item.def_id),
+                    sym::spotlight,
+                ),
```

Comment:
You already have the attributes available as `item.attrs`, you don't need `get_attrs`. That should also let you get rid of the new `has_doc_flag` function since you'll be working on `clean::Attributes`.

Reply:
Oh good catch! I'll remove this one. However, I still need this function because of `record_extern_trait` which, in certain contexts, doesn't have a `clean::Item` (like in `clean::utils::register_res`).